data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/PercentSamplerAggregateAction.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-Limense-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action 
+ * 
+ * @since 2.1
+ */
+@DataPrepperPlugin(name = "rate_limiter", pluginType = AggregateAction.class, pluginConfigurationType = PercentSamplerAggregateActionConfig.class)
+public class PercentSamplerAggregateAction implements AggregateAction {
+    static final String TOTAL_EVENTS_KEY = "total_events";
+    static final String TOTAL_ALLOWED_EVENTS_KEY = "total_allowed_events";
+    private final double percent;
+
+    @DataPrepperPluginConstructor
+    public PercentSamplerAggregateAction(final PercentSamplerAggregateActionConfig percentSamplerAggregateActionConfig) {
+        percent = percentSamplerAggregateActionConfig.getPercent();
+    }
+
+    @Override
+    public AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+        final GroupState groupState = aggregateActionInput.getGroupState();
+        if (groupState.get(TOTAL_EVENTS_KEY) == null) {
+            groupState.put(TOTAL_EVENTS_KEY, 1L);
+            groupState.put(TOTAL_ALLOWED_EVENTS_KEY, 0L);
```

Comment:
We probably still want to return an `AggregateActionResponse(event)` instead of the `nullEventResponse` for the case of the first event right?

Reply:
Good point. We want to return non null response for the first event only if the `percent` is 100. In all other cases, we will not allow the first event.